thorlcr/activities/xmlread/thxmlreadslave.cpp

```diff
@@ -96,12 +99,15 @@ class CXmlReadSlaveActivity : public CDiskReadSlaveActivityBase
         }
         virtual void close(CRC32 &fileCRC)
         {
-            CriticalBlock block(statsCs);
             xmlParser.clear();
             inputIOstream.clear();
             if (checkFileCrc)
                 fileCRC.reset(~crcStream->queryCrc()); // MORE should prob. change stream to use CRC32
-            mergeStats(fileStats, iFileIO);
+
+            {
+                CriticalBlock block(statsCs);
+                mergeStats(fileStats, iFileIO);
+            }
             iFileIO.clear();
```

Comment:
needs to be in the critical section (or assigned to a local variable outside - as in other places)